File: 프로그래머스/lv2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    static long[] cache;
    public long solution(int n) {
        // 피보나치
        cache = new long[n+1];
        long answer = dp(n);
    
        return answer;
    }
    
    static long dp(int n){
        cache[1] = 1;
        
        if(cache[n] != 0){
            return cache[n];
        }
        if(n == 2){
            cache[2] = 2;
        }
        if(n == 3){
            cache[3] = 3;
        }

        if(n >= 4){
            cache[2] = 2;
            cache[3] = 3;
            for(int i=4; i<=n ;i++){
                cache[i] = (cache[i-1] + cache[i-2])%1234567;
            }
        }

        return cache[n];
    }
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def solution(n):
    
    cache = [0]*n
    cache[0] = 1
    
    if n == 1:
        return cache[0]
    
    cache[1] = 2
    
    for i in range(2, len(cache)):
        cache[i] = cache[i-1] + cache[i-2]
    
    return cache[n-1]%1234567
